src/app/main.cpp

```diff
@@ -90,236 +92,139 @@ const char *sysSigName[] = {
+                if (app->isRunning()) {
+                    // Another instance had time to start.
+                    return EXIT_FAILURE;
+                }
+            }
+            else {
+                qCritical("Something went wrong while daemonizing, exiting...");
                 return EXIT_FAILURE;
             }
         }
-        else {
-            qCritical("Something went wrong while daemonizing, exiting...");
-            return EXIT_FAILURE;
-        }
-    }
 #else
-    if (!params.noSplash)
-        showSplashScreen();
+        if (!(params.noSplash || Preferences::instance()->isSplashScreenDisabled()))
+            showSplashScreen();
 #endif
 
 #if defined(Q_OS_UNIX) || defined(STACKTRACE_WIN)
-    signal(SIGINT, sigNormalHandler);
-    signal(SIGTERM, sigNormalHandler);
-    signal(SIGABRT, sigAbnormalHandler);
-    signal(SIGSEGV, sigAbnormalHandler);
+        signal(SIGINT, sigNormalHandler);
+        signal(SIGTERM, sigNormalHandler);
+        signal(SIGABRT, sigAbnormalHandler);
+        signal(SIGSEGV, sigAbnormalHandler);
 #endif
 
-    return app->exec(params.torrents);
-}
-
-QBtCommandLineParameters parseCommandLine()
-{
-    QBtCommandLineParameters result;
-    QStringList appArguments = qApp->arguments();
-
-    for (int i = 1; i < appArguments.size(); ++i) {
-        const QString& arg = appArguments[i];
-
-        if ((arg.startsWith("--") && !arg.endsWith(".torrent")) ||
-            (arg.startsWith("-") && arg.size() == 2)) {
-            //Parse known parameters
-            if ((arg == QLatin1String("-h")) || (arg == QLatin1String("--help"))) {
-                result.showHelp = true;
-            }
-#ifndef Q_OS_WIN
-            else if ((arg == QLatin1String("-v")) || (arg == QLatin1String("--version"))) {
-                result.showVersion = true;
-            }
-#endif
-            else if (arg.startsWith(QLatin1String("--webui-port="))) {
-                QStringList parts = arg.split(QLatin1Char('='));
-                if (parts.size() == 2)
-                    result.webUiPort = parts.last().toInt();
-            }
-#ifndef DISABLE_GUI
-            else if (arg == QLatin1String("--no-splash")) {
-                result.noSplash = true;
-            }
-#else
-            else if ((arg == QLatin1String("-d")) || (arg == QLatin1String("--daemon"))) {
-                result.shouldDaemonize = true;
-            }
-#endif
-            else {
-                //Unknown argument
-                result.unknownParameter = arg;
-                break;
-            }
-        }
-        else {
-            QFileInfo torrentPath;
-            torrentPath.setFile(arg);
-
-            if (torrentPath.exists())
-                result.torrents += torrentPath.absoluteFilePath();
-            else
-                result.torrents += arg;
-        }
+        return app->exec(params.torrents);
+    }
+    catch (CommandLineParameterError &er) {
```

Comment:
Agree, the body of `main()` is too long, we better split it onto several functions.